source/MaterialXGenGlsl/GlslShaderGenerator.cpp

```diff
@@ -770,6 +770,59 @@ ShaderNodeImplPtr GlslShaderGenerator::createCompoundImplementation(NodeGraphPtr
 }
 
 
+ValuePtr GlslShaderGenerator::remapEnumeration(const ValueElementPtr& input, const InterfaceElement& mappingElement, const TypeDesc*& enumerationType)
+{
+    const string& valueElementName = input->getName();
+    const string& valueString = input->getValueString();
+
+    enumerationType = nullptr;
+
+    ValueElementPtr valueElem = mappingElement.getChildOfType<ValueElement>(valueElementName);
+    if (!valueElem)
+    {
+        return nullptr;
+    }
+
+    const TypeDesc* inputTypeDesc = TypeDesc::get(input->getType());
+    // Don't convert file names and arrays to integers
+    if (inputTypeDesc->isArray() || inputTypeDesc == Type::FILENAME)
+    {
+        return nullptr;
+    }
+    // Don't convert supported types
+    if (getSyntax()->typeSupported(inputTypeDesc))
+    {
+        return nullptr;
+    }
+
+    // Skip any elements which have no enumerations
+    const string valueElemEnums = valueElem->getAttribute(ValueElement::ENUM_ATTRIBUTE);
+    if (valueElemEnums.empty())
+    {
+        return nullptr;
+    }
+
+    // Always update the type. For GLSL we always convert to integers,
+    // with the integer value being an index into the enumeration.
+    enumerationType = TypeDesc::get(TypedValue<int>::TYPE);
```

Comment:
Change so that type can change even when there is no value specified.
Swizzle strings are skipped here since they don't have a enumeration. They will remain strings and fall into the default conversion to int with a 0 value still (no change).